kafkabp/config.go

```diff
@@ -0,0 +1,81 @@
+package kafkabp
+
+import (
+	"github.com/Shopify/sarama"
+	"github.com/reddit/baseplate.go/log"
+)
+
+// ConsumerConfig can be used to configure a kafkabp Consumer.
+//
+// Can be deserialized from YAML.
+//
+// Example:
+//
+// kafka:
+//   brokers:
+//     - 127.0.0.1:9090
+//     - 127.0.0.2:9090
+//   topic: sample-topic
+//   clientID: myclient
+//   offset: oldest
+type ConsumerConfig struct {
+	// Required. Brokers specifies a slice of broker addresses.
+	Brokers []string `yaml:"brokers"`
+
+	// Required. Topic is used to specify the topic to consume.
+	Topic string `yaml:"topic"`
+
+	// Required. ClientID is a user-provided string sent with every request to
+	// the brokers for logging, debugging, and auditing purposes. The default
+	// Consumer implementation in this library expects every Consumer to have a
+	// unique ClientID.
+	//
+	// The Kubernetes pod ID is usually a good candidate for this unique ID.
+	ClientID string `yaml:"clientID"`
+
+	// Optional. Defaults to "oldest". Valid values are "oldest" and "newest".
+	Offset string `yaml:"offset"`
```

Comment:
nit: I almost forgot but I think we should define `"oldest"` and `"newest"` as exported string constants, so that users not using yaml can make sure they assign the correct value here. e.g.:

```go
// Allowed Offset values
const (
  OffsetOldest = "oldest"
  OffsetNewest = "newest"
)
```